src/main/webapp/components/workspaces/subscribe.js

```diff
@@ -30,6 +31,26 @@ const unsubscribeFromWorkspace = async props => {
   }
 }
 
+const onSubscribe = async (props, subscribed, setSubscribed) => {
+  const { subscribe, unsubscribe, id, title, setMessage } = props
+
+  subscribed
+    ? await unsubscribeFromWorkspace({
+        id,
+        title,
+        setMessage,
+        unsubscribe,
+        setSubscribed,
+      })
+    : await subscribeToWorkspace({
+        id,
+        title,
+        setMessage,
+        subscribe,
+        setSubscribed,
+      })
+}
+
 export default props => {
```

Comment:
:pencil2: Change this component to named export `SubscribeAction` to match other components